test/functional/tools/detect_errors.xml

```diff
@@ -0,0 +1,140 @@
+<tool id="detect_errors" name="detect_errors" version="1.0.0">
+    <stdio>
+       <exit_code range="3:5" level="warning" description="Low disk space" />
+       <exit_code range="6:"  level="fatal"   description="Bad input dataset" />
+       <exit_code range="2"   level="fatal_oom"   description="Out of Memory" />
+       <regex match="message" source="stdout" level="log" description="some program message of interest"/>
+       <regex match="low space" source="both" level="warning" description="Low space on device" />
+       <regex match="error" source="stderr" level="fatal" description="Unknown error encountered" />
+       <regex match="out of memory" source="stdout" level="fatal_oom" description="Out of memory error occurred" />
+       <regex match="[CG]{12}" description="Fatal error - CG island 12 nts long found" />
+       <regex match="^Branch A" level="warning" description="Branch A was taken in execution" />
+    </stdio>
+    <command><![CDATA[
+        #if str($stdoutmsg)!=""
+            echo "$stdoutmsg"; 
+        #end if
+        #if str($stderrmsg)!=""
+            >2& echo "$stderrmsg"; 
+        #end if
+        sh -c "exit $exit_code"
+    ]]>
+    </command>
+    <inputs>
+        <param name="stdoutmsg" type="text" value="" label="messages to stdout" />
+        <param name="stderrmsg" type="text" value="" label="messages to stderr" />
+        <param name="exit_code" type="integer" value="0" label="exit code"/>
+    </inputs>
+    <outputs>
+        <data name="out_file1" />
+    </outputs>
+    <tests>
+        <!-- exit code + regex tests -->
+        <!-- successful run (warning exit code + log and warning messages) -->
+        <test expect_exit_code="3" expect_failure="false">
+            <param name="stdoutmsg" value="message: everything is fine" />
+            <param name="stderrmsg" value="low space" />
+            <param name="exit_code" value="3" />
+            <assert_stdout>
+                <has_line line="Log: some program message of interest"/>
+                <has_line line="Warning: Low space on device"/>
+            </assert_stdout>
+            <assert_stderr>
+                <has_line line="Warning: Exit code 3 (Low disk space)"/>
+            </assert_stderr>
+        </test>
+        <!-- unccessful run (fatal exit code + warning in output messages are detected) -->
+        <test expect_exit_code="23" expect_failure="true">
+            <param name="stdoutmsg" value="Branch A seems to be of iterest .. maybe its related to the error?" />
+            <param name="exit_code" value="23" />
+            <assert_stdout>
+                <has_line line="Warning: Branch A was taken in execution"/>
+            </assert_stdout>
+(.venv) ➜  tools git:(topic/regex-for-oom) ✗ 
```

Comment:
Copy/paste mistake?

Reply:
Should be fine: 
- program writes "Branch A seems to be of iterest .. maybe its related to the error?" to stdout
- tool adds "Warning: Branch A was taken in execution" to stdout, and thats what I'm testing for here. 

But maybe I'm missing something?

Reply:
Look at https://travis-ci.org/galaxyproject/galaxy/jobs/393608375#L1180 ,  you cannot put text inside `<test>`, only tags.

Reply:
jep